core/src/main/java/org/elasticsearch/search/aggregations/InternalMultiBucketAggregation.java

```diff
@@ -82,6 +82,35 @@ public Object getProperty(List<String> path) {
         }
     }
 
+    /**
+     * Counts the number of inner buckets inside the provided {@link InternalBucket}
+     */
+    public static int countInnerBucket(InternalBucket bucket) {
+        int count = 0;
+        for (Aggregation agg : bucket.getAggregations().asList()) {
+            if (agg instanceof MultiBucketsAggregation) {
+                count += countInnerBucket((MultiBucketsAggregation) agg);
+            }
```

Comment:
I think we need to check single bucket aggregations here too? especially as they might contain more mutli-bucket aggregations below themZ

Reply:
Right, I pushed https://github.com/elastic/elasticsearch/pull/27581/commits/3fd25228e249a2a1f2d62d2710ecad3907da80f5